platform/linux-generic/odp_packet_io.c

```diff
@@ -485,6 +486,11 @@ int odp_pktio_start(odp_pktio_t hdl)
 		sched_fn->pktio_start(_odp_pktio_index(hdl), num, index, odpq);
 	}
 
+	if (ODP_PCAPNG) {
```

Comment:
Neat idea, but just put pcapng_prepare inside header and define it to empty val if ODP_PCAPNG is undefined.
Also if it returns an error, it should not require one to call pcapng_destroy() afterwards. Just make it clean up after itself in case of an error.

Reply:
And also it should be `_odp_pcapng_FOO`

Reply:
@lumag This style ensures that the code is always syntax-checked since it is compiled, but then optimized away if it is not active. I've found this to be a useful idiom.

Reply:
This is how dpdk zero-copy was introduced. I think we need to keep that for the same reasons @Bill-Fischofer-Linaro mentioned

Reply:
There is no need to syntax check function call, is it?
Anyway, `_prepare` call should not have side-effects in error case, so cleanup should be done internally.